### ISSUE TITLE
Query validation error updating Date field

### ISSUE THREAD

**Issue by kennytraction** [prisma/prisma#3512]
## Bug description

Attempting to update a Date field throws a query validation error after update to version 2.6.1

Error:
```
Invalid `prisma.entity.update()` invocation in\n/home/node/traction-data-prisma/src/prisma-entity-repository.ts:324:49\n\n\n  Failed to validate the query `Error occurred during query validation & transformation:\nMutation (object)\n  ↳ updateOneentity (field)\n    ↳ data (argument)\n      ↳ entityUpdateInput (object)\n        ↳ yearEndDate (field)\n          ↳ Value types mismatch. Have: String(\"2020-12-31T00:00:00.000Z\"), want: Object((Weak))` at `.Mutation.updateOneentity.data.entityUpdateInput.yearEndDate`
```

## How to reproduce
Given the following DB/Prisma schema, try to update the `yearEndDate` field.

Simplified PostgreSQL table:
```
CREATE TABLE IF NOT EXISTS "entity" (
  "id" uuid NOT NULL DEFAULT uuid_generate_v4(),
  "yearEndDate" date NOT NULL
  PRIMARY KEY ("id")
);
```
Schema Model:
```
model entity {
  id                        String             @default(dbgenerated()) @id
  yearEndDate               DateTime
}
```
Entity Update Type:
```
export type entityUpdateInput = {
  id?: string | StringFieldUpdateOperationsInput
  yearEndDate?: Date | string | DateTimeFieldUpdateOperationsInput
}
```
Update Entity:
```
this.client.entity.update({
            where: { id },
            data: {
                yearEndDate: input.yearEndDate
            }
        })
```
## Expected behavior

In version 2.5.1, the field was updated without an error.
Expect it to update without an error in version 2.6.1

## Environment & setup

- OS: Windows
- Database: PostgreSQL
- Node.js version: 12.16.1
- Prisma version: 2.6.1

**Comment by alan345** [comment]
Same issue here.

error: 
```
[31mInvalid [1m`ctx.prisma.authDevice.update()`[22m invocation in[39m [31m[4m/Users/alan/Documents/app/nacho/prismaTs/src/resolvers/subAction/manageAuthDevice.ts:62:55[24m[39m [2m [90m 58 [39mconst region = geo && geo.region ? geo.region : ''[22m [2m [90m 59 [39mconst eu = geo && geo.eu ? geo.eu : ''[22m [2m [90m 60 [39mconst city = geo && geo.city ? geo.city : ''[22m [2m [90m 61 [39m[22m [31m[1m→[22m[39m [2m[90m 62 [39mconst newAuthDevice = await ctx.prisma.authDevice.update([22m Failed to validate the query `Error occurred during query validation & transformation: Mutation (object) ↳ updateOneAuthDevice (field) ↳ data (argument) ↳ AuthDeviceUpdateInput (object) ↳ lastLogin (field) ↳ Value types mismatch. Have: String("2020-09-03T22:41:15.353Z"), want: Object((Weak))` at `.Mutation.updateOneAuthDevice.data.AuthDeviceUpdateInput.lastLogin`
```

code: 
```
    const newAuthDevice = await ctx.prisma.authDevice.update({
      data: {
        lastLogin: new Date(),
      },
      where: {
        id: authDevice.id,
      },
    })
```


In version 2.5.1, the field was updated without an error.


Environment & setup
OS: MAC
Database: MUSQL
Node.js version: v12.18.3
Prisma version: 2.6.1

I rollback to 2.5.1 for now.

**Comment by Jolg42** [comment]
Thank you for opening this issue, I'm looking into it and a patch will be available soon 👍

**Comment by Jolg42** [comment]
So we have a fix in our development build 2.6.2-dev.1 (not recommended for production usage) it will be part of the official 2.6.2 patch release soon.

**Comment by Jolg42** [comment]
`2.6.2` is released 😃 

https://github.com/prisma/prisma/releases/tag/2.6.2

**Comment by kennytraction** [comment]
@Jolg42 Confirmed this is fixed in 2.6.2. Thanks!!
